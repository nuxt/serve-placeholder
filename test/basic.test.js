const connect = require('connect')
const Axios = require('axios')
const placeholder = require('..')

describe('basic', () => {
  let app
  let server
  let axios

  it('init', () => {
    app = connect()

    app.use('/test', (req, res) => { res.end('Works!') })
    app.use(placeholder({}))

    server = app.listen(8080)
    axios = Axios.create({ baseURL: 'http://localhost:8080' })
  })

  it('/test', async () => {
    const response = await axios.get('/test')
    expect(response.data).toBe('Works!')
  })

  it('js', async () => {
    const response = await axios.get('/assets/foo.js').catch(e => e.response)
    expect(response.data).toMatchSnapshot()
  })

  it('json', async () => {
    const response = await axios.get('/assets/foo.json').catch(e => e.response)
    expect(response.data).toMatchSnapshot()
  })

  it('html', async () => {
    const response = await axios.get('/assets/foo.html').catch(e => e.response)
    expect(response.data).toMatchSnapshot()
  })

  it('css', async () => {
    const response = await axios.get('/assets/foo.css').catch(e => e.response)
    expect(response.data).toMatchSnapshot()
  })

  it('png', async () => {
    const response = await axios.get('/assets/foo.png').catch(e => e.response)
    expect(Buffer.from(response.data).toString('base64')).toMatchSnapshot()
  })

  it('close', () => {
    return new Promise(resolve => server.close(resolve))
  })
})
