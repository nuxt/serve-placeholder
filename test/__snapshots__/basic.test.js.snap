// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic css 1`] = `"/* style not found */"`;

exports[`basic html 1`] = `"<!-- page not found -->"`;

exports[`basic js 1`] = `"/* script not found */"`;

exports[`basic json 1`] = `Object {}`;

exports[`basic png 1`] = `"R0lGODlhAQABAO+/vQEA77+977+977+9AAAAIe+/vQQBCgABACwAAAAAAQABAAACAkwBADs="`;
